L/licensecheck/bundled/main.go

```diff
@@ -1,27 +1,123 @@
 package main
 
+/*
+#include <stdlib.h>
+*/
 import "C"
 import (
 	"github.com/google/licensecheck"
 
+	"log"
+	"sync"
 	"unsafe"
 )
 
+var (
+	_licenses []licensecheck.License
+	_scanner  *licensecheck.Scanner = nil
+	_mu       sync.Mutex
+)
+
 //export License
 func License(msg *C.char) (**C.char, int, float64) {
+    _mu.Lock()
+    if len(_licenses) == 0 {
+        _mu.Unlock()
+        return (**C.char)(nil), 0, 0.0
+    }
+    scanner := _scanner            // copy pointer under lock
+    _mu.Unlock()
+
+    if scanner == nil {
+        // shouldn't happen if you keep invariants, but be defensive
+        return (**C.char)(nil), 0, 0.0
+    }
+
 	bytes := []byte(C.GoString(msg))
-	cov := licensecheck.Scan(bytes)
+	cov := _scanner.Scan(bytes)
 
-	// https://stackoverflow.com/a/41493208
+    n := len(cov.Match)
+    if n == 0 {
+        return (**C.char)(nil), 0, cov.Percent
+    }
+	// Collect IDs from each match into a slice of *C.char
 	cArray := C.malloc(C.size_t(len(cov.Match)) * C.size_t(unsafe.Sizeof(uintptr(0))))
-	// mimic https://github.com/docker/docker-credential-helpers/pull/61
-	a := (*[(1 << 29) - 1]*C.char)(cArray)
 
-	for idx, m := range cov.Match {
-		a[idx] = C.CString(m.ID)
+	ids := (*[1<<30 - 1]*C.char)(cArray)[0:len(cov.Match)]
```

Comment:
this should fix the 32-bit build failure (with the other one below)
```suggestion
    ids := (*[1<<29 - 1]*C.char)(cArray)[:n:n]
```

Reply:
the original version of the wrapper had 1<<29 too, 1<<30 is too big for 32 bit builds. The slice thing is to set the size and capacity both to n, apparently that's a common idiom.